src/main/java/io/vertx/core/parsetools/impl/RecordParserImpl.java

```diff
@@ -210,9 +216,25 @@ private void handleParsing() {
       buff = EMPTY_BUFFER;
     } else {
       buff = buff.getBuffer(start, len);
+      if (streamEnded) {
```

Comment:
we should try to move this logic upper in the code block where there is:

```
        if (delimited) {
          next = parseDelimited();
        } else {
          next = parseFixed();
        }
```

so basically it would be something like:

```
        int next;
        if (delimited) {
          next = parseDelimited();
        } else {
          next = parseFixed();
        }
        Buffer event;
        if (next == -1) {
          if (streamEnded) {
            // ...
          } else {
            ReadStream<Buffer> s = stream;
            if (s != null) {
              s.resume();
            }
            break;
          }
        } else {
          event = buff.getBuffer(start, next);
        }
```

Reply:
What's wrong with the code? I would prefer to avoid a refactoring and simply patch 3.8. Then, as I said in comments earlier, rework the impl for 4.0.

Reply:
I believe we should not rework implemnetation for Vert.x 4 in order to make 3.x and 4 code similar to ease patch ports

Reply:
See my [initial comment](https://github.com/eclipse-vertx/vert.x/pull/2969#issuecomment-496454962)

If we fix this right, we will break compatibility

Reply:
actually we can't use inbound queue for such case because inbound queue requires to have a Vert.x context attached to it

Reply:
Right, this is why I proposed to leave the fix as-is for 3.x and revisit impl in 4.x

Reply:
you said you want to revisit using InboundQueue which is not feasible. How about fixing it the right away and we have deal done without requiring to revisit code that works ?

Reply:
Not feasible in 4.x?

Reply:
@vietj I've update the PR as requested